python/doc/examples/reliability_sensitivity/reliability/plot_nais.py

```diff
@@ -138,65 +144,46 @@ def fourBranch(x):
 )
 
 # %%
-# Draw the NAIS samples used by the algorithm
-# -------------------------------------------
-#
-# The following manipulations are possible only if you have created a :class:`~openturns.MemoizeFunction` that enables to store all the inputs and outputs of the function :math:`g`.
-#
-# Get all the inputs and outputs where :math:`g` were evaluated:
-
-# %%
-inputNAIS = g.getInputHistory()
-outputNAIS = g.getOutputHistory()
-nTotal = inputNAIS.getSize()
-print("Number of evaluations of g = ", nTotal)
+# You can also get the successive thresholds used by the algorithm:
 
 # %%
-# Within each step of the algorithm, a sample of size :math:`N` is created, where:
+levels = algo.getThresholdPerStep()
+print("Levels of g = ", levels)
 
 # %%
-N = algo.getMaximumOuterSampling() * algo.getBlockSize()
-print("Size of each subset = ", N)
+# Draw the NAIS samples used by the algorithm
+# ---------------------------------------------
 
 # %%
 # You can get the number :math:`N_s` of steps with:
+Ns = algo.getStepsNumber()
+print("Number of steps= ", Ns)
 
 # %%
-Ns = int(nTotal / N)
-print("Number of steps = ", Ns)
-
-# %%
-# Now, we can split the initial sample into NAIS samples of size :math:`N_s`:
-
-# %%
-listNAISSamples = list()
-listOutputNAISSamples = list()
-for i in range(Ns):
-    listNAISSamples.append(inputNAIS[i * N: i * N + N])
-    listOutputNAISSamples.append(outputNAIS[i * N: i * N + N])
-
-# %%
-# And get all the levels defining the intermediate and final thresholds given by the empirical quantiles of each NAIS output sample:
-
-# %%
-levels = []
-for i in range(Ns - 1):
-    levels.append(listOutputNAISSamples[i].computeQuantile(quantileLevel)[0])
-levels.append(threshold)
+# Get all the inputs where :math:`g` was evaluated at each step
+list_subSamples = list()
+for step in range(Ns):
+    list_subSamples.append(algo.getInputSample(step))
 
 # %%
 # The following graph draws each NAIS sample and the frontier :math:`g(x_1, x_2) = l_i` where :math:`l_i` is the threshold at the step :math:`i`:
 
 # %%
-graph = ot.Graph("NAIS samples", "x1", "x2", True, "lower left")
+graph = ot.Graph()
+graph.setAxes(True)
 graph.setGrid(True)
+graph.setTitle("NAIS sampling: samples")
+graph.setXTitle(r"$x_1$")
+graph.setYTitle(r"$x_2$")
+graph.setLegendPosition("lower left")
 
 # %%
 # Add all the NAIS samples:
 
 # %%
 for i in range(Ns):
-    cloud = ot.Cloud(listNAISSamples[i])
+    cloud = ot.Cloud(list_subSamples[i])
+    # cloud.setPointStyle("dot")
     graph.add(cloud)
 col = ot.Drawable().BuildDefaultPalette(Ns)
 graph.setColors(col)
```

Comment:
```suggestion
col = graph.getColors()
```
Default colors should be applied anyway. It is useful to save them here however, because you then want to add lines to your graph with the same colors.